crypto/threads_win.c

```diff
@@ -133,6 +133,18 @@ int CRYPTO_atomic_add(int *val, int amount, int *ret, CRYPTO_RWLOCK *lock)
     return 1;
 }
 
+int CRYPTO_atomic_read(int *val, int *ret, CRYPTO_RWLOCK *lock)
+{
+    InterlockedCompareExchange(val, 0, 0);
```

Comment:
And who updates *ret?

Reply:
Besides, if one argues that atomic_load is better than atomic_add with 0, because it translates to instruction with no lock prefix, then how come this one? This is as expensive instruction...

Reply:
I'm open to other suggestions as to how to go about this one Windows.
I don't know enough about the operating system.

Reply:
@paulidale, you forget the assignment to *ret, see [my original suggestion](https://github.com/openssl/openssl/pull/4414#discussion_r140693973 ).

I have no idea about the cost of these functions, @dot-asm seems to be better informed. If he says that the compare exchange is much more expensive than a simple attomic_add with 0, then you could use [InterlockedAdd()](https://msdn.microsoft.com/en-us/library/windows/desktop/ms683504.aspx) instead.

Reply:
Ok, sorry, I overlooked #4517.

Reply:
Yeah, I missed the assignment part of your suggestion @mspncp 
Final versions are best done by someone with substantial Windows experience.